src/envoy_cds.rs

```diff
@@ -53,15 +55,25 @@ impl tokio::stream::Stream for CDS {
 
     fn poll_next(
         mut self: std::pin::Pin<&mut Self>,
-        _: &mut Context<'_>,
-    ) -> Poll<Option<Result<DiscoveryResponse, tonic::Status>>> {
+        ctx: &mut Context<'_>,
+    ) -> Poll<Option<Self::Item>> {
+        let mut send_data = false;
         {
             let cfg = self.config.clone();
             let version = cfg.read().unwrap().get_version();
-            if self.version == version {
-                return Poll::Pending;
+            if self.version != version {
+                send_data = true;
             }
         }
+
+        if !(send_data) {
+            // @TODO config should return a future when a config is updated
+            log::trace!("Sleep CDS because no changes made");
```

Comment:
Just for our (including me!) information: the `tracing` crate is already a dependency of many tokio-related stuff we can use to replace the intended usages of `log`, with extras such as spans (and, I think, being able to pipe information to something like Jaeger should be doable once it is used).